lib/improver/nowcasting/accumulation.py

```diff
@@ -0,0 +1,209 @@
+
+        if fraction != 0:
+            msg = ("The specified accumulation period ({}) is not divisible "
+                   "by the time intervals between rates cubes ({}). As "
+                   "a result it is not possible to calculate the desired "
+                   "total accumulation period.".format(
+                       self.accumulation_period, time_interval))
+            raise ValueError(msg)
+
+        cube_subset = int(cube_subset)
+
+        # We expect 1 more rates cubes than accumulations cubes, if we have
+        # more or less, a targetted accumulation_period will be incomplete.
+        unused_cubes = int(len(cubes) % cube_subset)
+        if unused_cubes != 1 and cube_subset != 1:
+            msg = ("The provided cubes result in a partial period given the "
+                   "specified accumulation_period, i.e. the number of cubes "
+                   "is insufficient to give a set of complete periods. Only "
+                   "complete periods will be returned.")
+            warnings.warn(msg)
+
+        cube_subsets = []
+        for index in range(0, len(cubes) - unused_cubes, cube_subset):
+            cube_subsets.append(cubes[index:index + cube_subset + 1])
+        return cube_subsets
+
+    def process(self, cubes):
+        """
+        Calculate period precipitation accumulations based upon precipitation
+        rate fields.
+
+        Args:
+            cubes (iris.cube.CubeList):
+                A cubelist containing input precipitation rate cubes.
+        Returns:
+            accumulation_cubes (iris.cube.CubeList):
+                A cubelist containing precipitation accumulation cubes where
+                the accumulation periods are determined by plugin argument
+                accumulation_period.
+        """
+        # Standardise inputs to expected units.
+        cubes = enforce_coordinate_units_and_dtypes(
+            cubes, ['time', 'forecast_reference_time', 'forecast_period'],
+            inplace=False)
+        enforce_diagnostic_units_and_dtypes(cubes)
+
+        # Sort cubes into time order and calculate intervals.
+        cubes, times = self.sort_cubes_by_time(cubes)
+        try:
+            time_interval, = np.unique(np.diff(times, axis=0))
+        except ValueError:
+            msg = ("Accumulation is designed to work with "
+                   "rates cubes at regualar time intervals. Cubes "
+                   "provided are unevenly spaced in time; time intervals are "
+                   "{}.".format(np.diff(times, axis=0)))
+            raise ValueError(msg)
+
+        cube_subsets = self.get_period_sets(time_interval, cubes)
+
+        accumulation_cubes = iris.cube.CubeList()
+        for cube_subset in cube_subsets:
+            # Handle accumulation periods equal to the rates cube time interval
+            # in which case the last subset does not get used.
+            if len(cube_subset) == 1:
+                continue
```

Comment:
Do we have an agreement as to how accumulations will be calculated?  You've gone for a straight sum of rates: eg for an accumulation period of 0-15 minutes, sum the rates from 0-14 and ignore 15.  The alternative is to sum all 16 files, with half-weighting at 0 and 15.  This is how I've seen it done for radar, where each rate is valid at the exact time-stamp, and it's the same for the nowcast.  Otherwise you're essentially shifting your accumulation validity to be from -0.5 to 14.5, instead of 0 to 15.  Do you see?

Reply:
I've changed to use this as you suggest it is more standard.